File: PLP_Java/src/Aula03_ListaDeExercicio/Q14_Configuracao.java
package Aula03_ListaDeExercicio;

class Q14_Configuracao {
    private static Q14_Configuracao instance;
    private String setting;

    // Construtor privado para impedir a criação de múltiplas instâncias
    private Q14_Configuracao() {
        this.setting = "Configuração padrão";
    }

    // Método estático que retorna a única instância
    public static Q14_Configuracao getInstance() {
        if (instance == null) {
            instance = new Q14_Configuracao();
        }
        return instance;
    }

    // Getters e setters
    public String getSetting() {
        return setting;
    }

    public void setSetting(String setting) {
        this.setting = setting;
    }

    public static void main(String[] args) {
        Q14_Configuracao config1 = Q14_Configuracao.getInstance();
        Q14_Configuracao config2 = Q14_Configuracao.getInstance();

        System.out.println(config1 == config2);
        config1.setSetting("Nova Configuração");
        System.out.println(config2.getSetting());
    }
}

File: PLP_python/Aula03_ListaDeExercicios(15)/Q14_Configuracao.py
class Q014_Configuracao:
    _instance = None

    def __new__(cls):
        if cls._instance is None:
            cls._instance = super(Q014_Configuracao, cls).__new__(cls)
            cls.setting = "Configuração padrão"
        return cls._instance

if __name__ == "__main__":
    config1 = Q014_Configuracao()
    config2 = Q014_Configuracao()

    print(config1 is config2)
    config1.setting = "Nova Configuração"
    print(config2.setting)
